libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -8,15 +13,72 @@
  */
 #define MQTT_NEXT_PACKET_ID_START    ( 1 )
 
+/**
+ * @brief A PINGREQ packet is always 2 bytes in size, defined by MQTT 3.1.1 spec.
+ */
+#define MQTT_PACKET_PINGREQ_SIZE     ( 2U )
+
+/**
+ * @brief A packet type not handled by MQTT_ProcessLoop.
+ */
+#define MQTT_PACKET_TYPE_INVALID     ( 0U )
+
+/**
+ * @brief Number of milliseconds in a second.
+ */
+#define MQTT_ONE_SECOND_TO_MS        ( 1000U )
+
+/**
+ * @brief Zero timeout in the process loop implies one iteration.
+ */
+#define MQTT_NO_TIMEOUT_MS           ( 0U )
+
+/**
+ * @brief Length of time spent for single test case with
+ * multiple iterations spent in the process loop for coverage.
+ */
+#define MQTT_SAMPLE_TIMEOUT_MS       ( 1U )
+
+/**
+ * @brief Length of the MQTT network buffer.
+ * 1232 bytes is the size of the largest packet (ACK) for running unit tests.
+ */
+#define MQTT_TEST_BUFFER_LENGTH      ( 1232 )
+
+/**
+ * @brief The packet type to be received by the process loop.
+ * IMPORTANT: Make sure this is set before calling expectProcessLoopCalls(...).
+ */
+static uint8_t currentPacketType = MQTT_PACKET_TYPE_INVALID;
+
+/**
+ * @brief The return value of modifyIncomingPacket(...) CMock callback that
+ * replaces a call to MQTT_GetIncomingPacketTypeAndLength.
+ * IMPORTANT: Make sure this is set before calling expectProcessLoopCalls(...).
+ */
+static MQTTStatus_t modifyIncomingPacketStatus = MQTTSuccess;
+
+/**
+ * @brief Time at the beginning of each test. Note that this is not updated with
+ * a real clock. Instead, we simply increment this variable.
+ */
+static uint32_t globalEntryTime = 0;
+
+/**
+ * @brief A static buffer used by the MQTT library for storing packet data.
+ */
+static uint8_t mqttBuffer[ MQTT_TEST_BUFFER_LENGTH ] = { 0 };
```

Comment:
Do we ever even write into this buffer? It seems to me all the serialize calls that would do so are mocked out, as well as the transport receive.

Reply:
It's not needed but a buffer of sufficient size is needed for the parameter checks to pass, and I think you're using the buffer in #964 so I'll just leave this variable declared.

Reply:
If the unit tests can run with smaller memory footprint, that does seem desirable, given our target platforms.

Reply:
Removed the static buffer